### DISCUSSION TITLE
Laravel Socialite provider example

### DISCUSSION THREAD

**Discussion by binaryfire** [zitadel/zitadel#4526] [Ideas]
Hi guys,

Would you consider adding a Laravel Socialite example to https://github.com/zitadel/zitadel-examples? Laravel is an extremely popular backend framework: https://statisticsanddata.org/data/most-popular-backend-frameworks-2012-2022/

Socialite is the core Laravel package used for IDP authentication: https://laravel.com/docs/9.x/socialite

These are the community-maintained third party providers: https://socialiteproviders.com/

Here's some info on creating custom providers:
https://github.com/SocialiteProviders/Manager
https://medium.com/laravel-news/adding-auth-providers-to-laravel-socialite-ca0335929e42

And this is the Keycloak provider:
https://github.com/SocialiteProviders/Providers/tree/master/src/Keycloak

**Comment by mffap** [comment]
Hey @binaryfire - I like the idea. As it seems we could take many ideas from already existing providers such as Keycloak or Okta. Both look quite comprehensive.

 I've used Laravel quite some time ago. When I get some time, I can have a look. Posting here is very useful, maybe someone else from the community has experience with it and would be able to contribute. 

Thanks for raising the Discussion.

**Reply by binaryfire** [reply]
@mffap Awesome. Unfortunately my coding skills aren’t good enough to contribute but I’ll spread the word once a solution is available :)
